### PR TITLE
Reflecting the case-insensitivity of http headers

### PR DESCRIPTION
**Pull request by therealryan**
I recently went on a really unfun yak-shaving expedition when I:
 1. Added an endpoint that uses the `PATCH` http method
 2. Added a flow that exercised it
 3. Discovered that the standard java `HttpURLConnection` stuff just [doesn't support that](https://stackoverflow.com/questions/25163131/httpurlconnection-invalid-http-method-patch)
 4. Refactored to use the new [`HTTPClient` stuff](https://docs.oracle.com/en/java/javase/11/docs/api/java.net.http/java/net/http/HttpClient.html) in java 11
 5. Found that [`HTTPHeaders`](https://docs.oracle.com/en/java/javase/11/docs/api/java.net.http/java/net/http/HttpHeaders.html) only reports lower-case header names, even though we're confident that the service returns mixed-case. This broke the `chunked` transfer-encoding stuff that the `HttpMsg` types do, and required a bunch of tedious test-data updates.

`HTTPHeaders`'s casing behaviour is the [correct position](https://www.rfc-editor.org/rfc/rfc9110.html#section-5.1-2), so we're going to adapt to that.

### CODE REVIEW

**Comment by qianliok** [comment]
test?

File: message/message-http/src/main/java/com/mastercard/test/flow/msg/http/HttpMsg.java [hunk 1]
```diff
@@ -248,16 +248,24 @@ protected static boolean isHeaderField( String field ) {
...
+	 * supplied name to lower-case before adding the sigils that identify it as a
+	 * header field.
+	 *
 	 * @param name The desired header name
 	 * @return The value to pass as the field name in {@link #set(String, Object)}
 	 *         and {@link #get(String)}
 	 */
 	public static String header( String name ) {
-		return HEADER_PRESUFIX + name + HEADER_PRESUFIX;
+		return HEADER_PRESUFIX + name.toLowerCase() + HEADER_PRESUFIX;
```
Comment (therealryan) [inline message/message-http/src/main/java/com/mastercard/test/flow/msg/http/HttpMsg.java:262]: This is the key change - we now convert all header names to lower case when they get added to the message data model.

File: message/message-http/src/main/java/com/mastercard/test/flow/msg/http/HttpReq.java [hunk 2]
```diff
@@ -75,8 +75,7 @@ public HttpReq( byte[] content, Function<byte[], Message> bodyParse ) {
 		while( !lines.isEmpty() && !(headerLine = lines.removeFirst()).isEmpty() ) {
 			Matcher h = HEADER_LINE_PATTERN.matcher( headerLine );
 			if( h.matches() ) {
-				set( HEADER_PRESUFIX + h.group( "name" ).trim() + HEADER_PRESUFIX,
-						h.group( "value" ).trim() );
+				set( header( h.group( "name" ).trim() ), h.group( "value" ).trim() );
```
Comment (therealryan) [inline message/message-http/src/main/java/com/mastercard/test/flow/msg/http/HttpReq.java:78]: Removing these ad-hoc reimplementations of the `header` method in favour of just using the real thing

File: message/message-http/src/main/java/com/mastercard/test/flow/msg/http/HttpMsg.java (hunk 1 above)
Comment (oconnord) [reply message/message-http/src/main/java/com/mastercard/test/flow/msg/http/HttpMsg.java:262]: Why the prefix/suffix on headers? I'm not familiar with it being part of the specification.

**Review by apanayotopoulos — APPROVED** [review]
Looks pretty straightforward to me

**Review by oconnord — APPROVED** [review]
Happy with these changes.

**Review by pratyp — APPROVED** [review]

**Pull request merged**